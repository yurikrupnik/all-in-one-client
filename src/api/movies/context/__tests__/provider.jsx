
import React from 'react';
import Provider, { MoviesProviderAsClass } from '../provider';
import { Provider as P } from '../index';
// import api from '../../api';


// jest.mock('../../api'); // eslint-disable-line no-undef

const {
    describe,
    test,
    expect,
    shallow
} = global;

describe('movies provider', () => {
    const Component = () => (
        <div>
            helo
        </div>
    );

    test('renders movies Provider', () => {
        const wrapper = shallow((
            <Provider>
                <Component />
            </Provider>
        ));
        expect(wrapper).toMatchSnapshot();
        // expect(1).toBe(1)
    });

    test('should test fetch', () => {
        // expect(1).toBe(1);
        const wrapper = shallow((
            <MoviesProviderAsClass>
                <Component />
            </MoviesProviderAsClass>
        ));
        const instance = wrapper.instance();
        instance.setSelected({ id: 123 });
        instance.setData([{ id: 123 }]);
        instance.setSearch('oz');
        instance.toggleOpen();
        // expect(api.fetch).toHaveBeenCalledTimes(1);
        // console.log('instance.state', instance.state);
        // return done();
    });

    test('renders P', () => {
        const wrapper = shallow((
            <P>
                <Component />
            </P>
        ));
        expect(wrapper).toMatchSnapshot();
        // expect(1).toBe(1)
    });
});
