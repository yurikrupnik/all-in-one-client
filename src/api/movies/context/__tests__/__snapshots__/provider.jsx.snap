// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movies provider renders P 1`] = `
<ContextProvider
  value={
    Object {
      "handleChange": [Function],
      "handleSelect": [Function],
      "open": false,
      "search": "",
      "selected": Object {},
      "setData": [Function],
      "shows": Array [],
      "toggleOpen": [Function],
    }
  }
>
  <Component />
</ContextProvider>
`;

exports[`movies provider renders movies Provider 1`] = `
<ContextProvider
  value={
    Object {
      "handleChange": [Function],
      "handleSelect": [Function],
      "open": false,
      "search": "",
      "selected": Object {},
      "setData": [Function],
      "shows": Array [],
      "toggleOpen": [Function],
    }
  }
>
  <Component />
</ContextProvider>
`;
